backend/src/main/scala/cromwell/backend/ReadLikeFunctions.scala

```diff
@@ -2,13 +2,22 @@ package cromwell.backend
 
 import cromwell.core.io.AsyncIoFunctions
 import cromwell.core.path.PathFactory
+import cromwell.filesystems.drs.{DrsPath, DrsResolver}
 import wom.expression.IoFunctionSet
 
 import scala.concurrent.Future
 import scala.util.Try
 
 trait ReadLikeFunctions extends PathFactory with IoFunctionSet with AsyncIoFunctions {
 
+  override def resolvedFileBasename(value: String): Future[String] = buildPath(value) match {
+    case drsPath: DrsPath => DrsResolver.getResolvedBasename(drsPath).unsafeToFuture()
+    case path =>
+      val name = path.name
+      if (name.nonEmpty) Future.successful(name)
+      else Future.failed(new Exception(s"No resolvable basename for $value. Was it a directory?"))
```

Comment:
I don't know enough about `BetterFileMethods.name` to be certain that this is equivalent to `fileNameAsString.valueString.split('/').last`. I'm willing to trust it, though, if you can confirm that that is the intent.

Reply:
Yes, except that the former requires a valid path (as above) whereas the string split/replace will always give you _something_ even if the input is not a meaningful path